webpack/components/NewTemplateSync/NewTemplateSyncActions.js

```diff
@@ -0,0 +1,32 @@
+import api from 'foremanReact/API';
+
+import {
+  SYNC_SETTINGS_REQUEST,
+  SYNC_SETTINGS_SUCCESS,
+  SYNC_SETTINGS_FAILURE,
+} from '../../consts';
+
+import { deepPropsToCamelCase } from '../../helpers';
+
+export const getSyncSettings = url => dispatch => {
+  dispatch({ type: SYNC_SETTINGS_REQUEST });
+  return api
```

Comment:
Would you mind migrating to `async/await` syntax?

It is recommended because we already started the work on the migration:
https://github.com/theforeman/foreman/pull/6901

It's include a new `eslint` rule that will block you from using `.then/.catch` and we will include it in the `@theforeman/env` package so it is better to migrate now so you won't get the errors later.

I would recommend using the eslint plugin `promise` as @amirfefer use in his PR.